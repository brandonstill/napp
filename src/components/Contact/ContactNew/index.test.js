import React from 'react';
import { shallow, mount } from 'enzyme';
import { ContactNew } from './';
import Button from '@material-ui/core/Button';

describe('ContactNew', () => {
  
  it('renders correctly', () => {
    const wrapper = shallow(<ContactNew classes={{container:''}} />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it('has correct default state', () => {
    const wrapper = shallow(<ContactNew classes={{container:''}} />).instance();
    const defaultState = { 
      firstName: '',
      lastName: '',
      email: '',
      firstNameError: false,
      lastNameError: false,
      emailError: false
    };
    expect(wrapper.state).toEqual(defaultState);
  });

  it('handleChange updates component state', () => {
    const wrapper = shallow(<ContactNew classes={{container:''}} />);
    wrapper.find('TextField[name="firstName"]').simulate('change', { target: { value: 'First Name' } });
    expect(wrapper.state('firstName')).toBe('First Name');
  });

  it('handleSubmit call handleSave action with component state', () => {
    const state = { 
      firstName: 'First Name',
      lastName: 'Last Name',
      email: 'email@email.com'
    };
    const event = { preventDefault: jest.fn() };
    const spy = jest.fn();
    const wrapper = shallow(<ContactNew handleSave={spy} classes={{container:''}} />);
    wrapper.setState(state);
    wrapper.instance().handleSubmit(event);
    expect(spy).toHaveBeenCalledWith(state);
  });

  it('Cancel button calls handleClose', () => {
    const spy = jest.fn();
    const wrapper = mount(<ContactNew handleClose={spy} classes={{container:''}} />);
    wrapper.find('Button[name="cancel"]').simulate('click');
    expect(spy).toHaveBeenCalled();
  });

});