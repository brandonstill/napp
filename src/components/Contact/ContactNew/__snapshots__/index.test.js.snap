// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactNew renders correctly 1`] = `
Array [
  <form
    autoComplete="off"
    className=""
    noValidate={true}
    onSubmit={[Function]}
>
    <WithStyles(Typography)
        component="h2"
        gutterBottom={true}
        variant="headline"
    >
        Add New Contact
    </WithStyles(Typography)>
    <TextField
        className={undefined}
        error={false}
        id="firstName"
        label="First Name"
        margin="normal"
        name="firstName"
        onChange={[Function]}
        required={true}
        select={false}
        style={
            Object {
                "marginLeft": 0,
                "marginRight": 0,
              }
        }
        value={undefined}
        variant="standard"
    />
    <TextField
        className={undefined}
        error={false}
        id="lastName"
        label="Last Name"
        margin="normal"
        name="lastName"
        onChange={[Function]}
        required={true}
        select={false}
        style={
            Object {
                "marginLeft": 0,
                "marginRight": 0,
              }
        }
        value={undefined}
        variant="standard"
    />
    <TextField
        className={undefined}
        error={false}
        id="email"
        label="Email"
        margin="normal"
        name="email"
        onChange={[Function]}
        required={true}
        select={false}
        style={
            Object {
                "marginLeft": 0,
                "marginRight": 0,
              }
        }
        value={undefined}
        variant="standard"
    />
    <div
        className={undefined}
    >
        <WithStyles(Button)
            className={undefined}
            name="cancel"
            onClick={undefined}
            size="large"
            type="button"
        >
            Cancel
        </WithStyles(Button)>
        <WithStyles(Button)
            color="primary"
            name="submit"
            size="large"
            type="submit"
            variant="contained"
        >
            Add
        </WithStyles(Button)>
    </div>
</form>,
]
`;
